hack/update-versions.sh

```diff
@@ -0,0 +1,51 @@
+#! /usr/bin/env bash
+
+# Copyright 2022 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+CURRENT_RELEASE=${CURRENT_RELEASE:-1.5}
+PREVIOUS_RELEASE=${PREVIOUS_RELEASE:-1.4}
+
+
+echo "Updating kn from $PREVIOUS_RELEASE to $CURRENT_RELEASE"
+newfile="kn.rb"
+oldfile="kn@${PREVIOUS_RELEASE}.rb"
+
+echo "Create formula for previous release"
+cp "$newfile" "$oldfile"
+class_version=${PREVIOUS_RELEASE//.}
+class_name="Kn"
+sed -i 's/class '"${class_name}"' < Formula/class '"${class_name}"'AT'"$class_version"' < Formula/' "$oldfile"
+echo "$oldfile created"
+
+echo "Creating formula for current release"
+sed -i 's/v'"${PREVIOUS_RELEASE}"'/v'"$CURRENT_RELEASE"'/g' "$newfile"
+
+# change shas for mac and linux
+checksums=$(mktemp)
+old_checksums=$(mktemp)
+curl -L "https://github.com/knative/client/releases/download/knative-v${CURRENT_RELEASE}.0/checksums.txt" > "$checksums"
+curl -L "https://github.com/knative/client/releases/download/knative-v${PREVIOUS_RELEASE}.0/checksums.txt" > "$old_checksums"
```

Comment:
It does not appear the formulas here have ever been updated for a patch release. Easiest thing is to probably just include the patch level in the release number. I think that's fine for the client, but for the plugins brew repo it'll be trickier, as plugins can vary on if/when they release patches. 

Part of me wouldn't be too upset if we just dropped all the homebrew stuff...

Reply:
Thinking it through a bit more, given that we haven't previously been updating patch releases, I think I'd rather not tackle that complexity now. 

If we do decide to start updating patch releases, I think it'd be better to move to the formulas to the [template used by the event plugin](https://github.com/knative-sandbox/homebrew-kn-plugins/blob/main/event.rb), which doesn't require updating checksums in the formulas. Once we make that move, we could drop the checksum parts of the script completely.

Reply:
> If we do decide to start updating patch releases, I think it'd be better to move to the formulas to the [template used by the event plugin](https://github.com/knative-sandbox/homebrew-kn-plugins/blob/main/event.rb), which doesn't require updating checksums in the formulas. Once we make that move, we could drop the checksum parts of the script completely.

We briefly touched that topic on the event plugin PR. There was a security concern raised by Roland, since dynamically resolved checksums defeat the purpose of verifying the origin. On other hand to make the integrity check more robust we would need to sign our binaries/releases in general.